discovery-provider/src/premium_content/signature.py

```diff
@@ -38,3 +42,37 @@ def get_premium_content_signature(
     }
     signature = generate_signature(data)
     return {"data": cast(PremiumContentSignatureData, data), "signature": signature}
+
+
+# Subclass JSONEncoder
```

Comment:
this block and the generate_signature below are duplicated from another file, but imported from there causes some circular dependency that I haven't yet worked out.
going to find a way to use existing functions without circular dep

Reply:
done